cint/cint/src/loadfile.cxx

```diff
@@ -559,6 +559,46 @@ int G__isfilebusy(int ifn)
   return(flag);
 }
 
+#if !defined(G__WIN32)
+
+#include <errno.h>
+#include <time.h>
+
+struct G__StatCacheEntry {
+  struct stat info;
+  time_t ctime;
+  int retcode, _errno;
+};
+
+#ifndef R__STAT_CACHING_TIME
+#define R__STAT_CACHING_TIME    60
+#endif
+
+int G__cachingstat(const char *path, struct stat *buf) {
+  static std::map<std::string, struct G__StatCacheEntry> stat_cache;
+
+  if (   stat_cache.count(path)
```

Comment:
Would it be possible to use the result of `std::map::find` rather than searching for the location 3 times (for the case where the path is found)?

Reply:
We could, but I'm not sure if it's worth complicating the code for optimizing without benchmarks.

Looking at optimized build assembly[1], there aren't any separate calls to `std::map::operator[]` associated with the condition or the body of this if.

[1] https://godbolt.org/z/EdzrG4enh

Reply:
> Looking at optimized build assembly[1], there aren't any separate calls to `std::map::operator[]` associated with the condition or the body of this if.
> 
> [1] https://godbolt.org/z/EdzrG4enh

On a second look, I'm wrong, they are there. I will update the code to use find.

Reply:
Done, although I doubt it will have any effect on performance.